server/routes/blog.service.js

```diff
@@ -25,29 +26,50 @@ const getBlogByCategory = async (category) => {
 
 // mutations
 const createBlog = async (blogData) => {
+
+    const validatedBlog = BlogSchema.parse(blogData)
+
     return db.blog.create({
-        data: {
-            title: blogData.title,
-            category: blogData.category,
-            description: blogData.description,
-            content: blogData.content,
-            imageUrl: blogData.imageUrl,
-            authorId: blogData.authorId,
-            datePublished: blogData.datePublished,
-        },
+        data:validatedBlog,
         select: {
             id: true,
         },
     })
 }
 
 const updateBlog = async (id, blogData) => {
+    const validatedData = BlogSchema.parse(blogData)
```

Comment:
Try using safeParse() instead of parse(). Then inside of an if statement, do:
`if (!validatedBlog.success) {
  // return the error however you want and then close the function
}`

With the regular parse, a plain error would just be thrown, you have to handle it with a try and catch and then figure out if it was a zodError and so on. With safeParse, you can just catch the error within the if statement and then deal with it right on the spot as soon as you catch it.

Reply:
instead of this, how about I create a `try` `catch` instead?
something like this:
```javascript
const createBlog = async (blogData) => {
    const validatedBlog = BlogSchema.parse(blogData)

    try {
        return db.blog.create({
            data: validatedBlog,
            select: {
                id: true,
            },
        })
    } catch (error) {
        throw new Error(error)
    }
}```

Reply:
The try and catch should include the validation part if you prefer using it.

But I personally like safeParse because then you don't need a try and catch. From what I've seen when doing research behind this, try and catches are actually less performant than if/else statements.

Reply:
understood, what do you think about this:
```javascript
const createBlog = async (blogData) => {
    const validatedBlog = BlogSchema.safeParse(blogData)

    if (!validatedBlog.success) {
        throw new Error("Invalid blog data")
    } else {
        return db.blog.create({
            data: validatedBlog.data,
            select: {
                id: true,
            },
        })
    }
}
```

Reply:
That's good. Only thing I would change is get rid of the else block. No need for an else block here as the function would automatically end if the error is thrown.